### PR TITLE
Updated README

### PR DESCRIPTION
**Pull request by dilan-dio4**

### CODE REVIEW

**Comment by dilan-dio4** [comment]
@porsager Here's where I'm at with restructuring some of the README. Let me know your thoughts.

**Comment by porsager** [comment]
👏 I love it 😊 

cursor now also returns an async iterable if called without a callback function. 

I'll comment on the rest in line.

File: README.md [hunk 1]
```diff
@@ -1,132 +1,109 @@
...
 - 🚯 1850 LOC - 0 dependencies
 - 🏷 ES6 Tagged Template Strings at the core
 - 🏄‍♀️ Simple surface API
+- 🖊️ Dynamic query support
 - 💬 Chat on [Gitter](https://gitter.im/porsager/postgres)
 
 <br>
 
+# Table of Contents
+
```
Comment (porsager) [inline README.md:13]: Should we perhaps move the table of contents below getting started?

File: README.md [hunk 2]
```diff
@@ -145,378 +122,349 @@ const [new_user] = await sql`
...
-sql`
-  update users set ${
-    sql(user, 'name')
-  } where 
-    id = ${ user.id }
+const users = await sql`
+  select
+    *
+  from users
+  where age in (${ [68, 75, 23] })
```
Comment (porsager) [inline README.md:345]: This is deprecated in v3. It's done like this now by removing parenthesis and using sql():
```
in ${ sql(array) }
```

**Review by porsager — COMMENTED** [review]
Ok, that's as far as i got on mobile, I have some more input, but will have to wait until I'm at my computer. 😊

File: README.md (hunk 1 above)
Comment (dilan-dio4) [reply README.md:13]: Yes, will do

File: README.md [hunk 3]
```diff
@@ -389,30 +389,23 @@ await sql`
 Use cursors if you need to throttle the amount of rows being returned from a query. New results won't be requested until the promise / async callback function has resolved.
 
 ```js
-
-await sql`
-  select * from generate_series(1,4) as x
-`.cursor(async ([row]) => {
+for await (const row of sql`select * from generate_series(1,4) as x`.cursor()) {
```
Comment (porsager) [inline README.md:390]: It's also always a result array being returned, even if a single row, so it needs to be `const [row] of ...`

**Comment by porsager** [comment]
Ok, going a bit more thoroughly through it here is a list of some more things missing

- [ ] ``` const stream = await sql`copy x from stdin`.writable() ``` and ``` const stream = await sql`copy x to stdout`.readable() ```  is missing.

- [ ] Would be great with a description of the default `max_lifetime` setting which is a random interval for each connect between 45 and 90 minutes. This seemed to me like the best tradeoff between gaining the benefits of prepared statements, but still staying good friends with the OOM killer on linux. The rand9om interval was set to avoid connections reconnecting in bulk at the exact same time.

- [ ] Large objects usage documentation is not a must for v3 release, but would be nice.

- [ ] A description of the fact that queries are not executed until awaited or `.execute()` is called. This is necessary to support the usage of ``` sql`` ``` for fragments

- [ ] A mention of Deno support perhaps

File: README.md (hunk 3 above)
Comment (dilan-dio4) [reply README.md:390]: Shoot I missed that lol

Comment (porsager) [reply README.md:390]: Hehe.. that's on me for just making these things with no traces ;)

**Comment by dilan-dio4** [comment]
Changes made. I'll leave `stream` and Deno to you since I'm not too familiar in those spaces.

**Comment by porsager** [comment]
Perfect, I think the rest should be manageable for me one of the coming days. 

Do you have more changes, or is merging fine now?

**Comment by dilan-dio4** [comment]
Merging is good for now. If I have some more ideas over the next few days I'll add them as well.

**Comment by porsager** [comment]
Awesome! Thanks again :)

**Comment by dilan-dio4** [comment]
Of course! Do we have an ETA for v3 release?

**Comment by porsager** [comment]
I don't have any outstanding issues now, so once I've added the rest to the docs we're good to go. During the coming week seems possible.

**Comment by dilan-dio4** [comment]
Awesome!

**Pull request merged**